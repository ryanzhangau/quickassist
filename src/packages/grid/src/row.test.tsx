import React from 'react'
import { render } from '@testing-library/react'
import AnctechRow, { generateClass, ColsForRow, Halign, VAlignment } from './row'

describe('AnctechRow component', () => {
  it('should match <AnctechRow /> snapshot', () => {
    const { asFragment } = render(<AnctechRow>test</AnctechRow>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should generate cols per row class when colsPerRow is provided', () => {
    const colsPerRow: ColsForRow = { xs: 3, lg: 4 }
    expect(generateClass({ colsPerRow })).toBe('row row-cols-xs-3 row-cols-lg-4')
  })

  it('should generate horizontal aligment class when hAlign is proived', () => {
    const hAlign: Halign = { xs: 'start', lg: 'end' }
    expect(generateClass({ hAlign })).toBe('row justify-content-xs-start justify-content-lg-end')
  })

  it('should generate vertical alignment class when vAlgin is provide', () => {
    const vAlign: VAlignment = 'start'
    expect(generateClass({ vAlign })).toBe('row align-items-start')
  })
})
