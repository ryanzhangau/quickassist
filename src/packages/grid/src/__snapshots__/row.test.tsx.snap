// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnctechRow component should match <AnctechRow /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    test
  </div>
</DocumentFragment>
`;
